Tribler/Core/Utilities/utilities.py

```diff
@@ -109,94 +179,164 @@ def validTorrentFile(metainfo):
         l = info['length']
         if not isinstance(l, IntType) and not isinstance(l, LongType):
             raise ValueError('info length is not int, but ' + repr(type(l)))
-    else:
+    elif 'files' in info:
         # multi-file torrent
         files = info['files']
         if not isinstance(files, ListType):
             raise ValueError('info files not list, but ' + repr(type(files)))
 
         filekeys = ['path', 'length']
         for file in files:
-            for key in filekeys:
-                if key not in file:
-                    raise ValueError('info files missing path or length key')
+            if not all(key in file for key in filekeys):
+                raise ValueError('info files missing path or length key')
 
             p = file['path']
             if not isinstance(p, ListType):
                 raise ValueError('info files path is not list, but ' + repr(type(p)))
-            for dir in p:
-                if not isinstance(dir, StringType):
-                    raise ValueError('info files path is not string, but ' + repr(type(dir)))
+
+            if not all(isinstance(dirP, StringType) for dirP in p):
+                raise ValueError('info files path is not string')
 
             l = file['length']
             if not isinstance(l, IntType) and not isinstance(l, LongType):
                 raise ValueError('info files length is not int, but ' + repr(type(l)))
+    else:
+        raise ValueError("neither length or files found in files information")
+    return None
+
+
+def validate_torrent_info(metainfo):
+    """
+    Validate the info in the metainfo if it exists.
+
+    The following information is validated:
+        -   Whether an info key exists within metainfo
+        -   Whether necessary keys exist within information (name, piece length and either root hash or pieces)
+        -   Whether the name value is of type String
+        -   Whether the piece length value is of type Long or Integer
+        -   Whether the root hash is an instance of String qnd has a length of 20 if it exists
+        -   Whether the pieces value is an instance of String and it has a length of multiple of 20
+
+    The validate_files function is used to validate more as well.
```

Comment:
This is not very descriptive. I don't think it's necessary to mention the `validate_files` method here.